WordPress/wordpress-playground
# [Analytics] Track install and error events

This PR will track which plugins and themes were installed during boot from a Blueprint, by sending a `installTheme` and `installPlugin` events. 
An `error` event will be sent if Playground experiences a boot or PHP request error.

All events are sent to Google Analytics and don't contain any personal data information. 

`InstallTheme` and `InstallPlugin` events will include the resource type and the theme/plugin slug if the resource type is WordPress.org.

`Error` events will only send an error `source` which can be:

- `bootSiteClient` - error occurred during boot.
- `request` - a PHP or SQL error occurred during a PHP request.
- `unknown` - for request errors that don't have a source. This shouldn't happen but we still want to catch it to confirm it's not happening in practice. 

The purpose of the `error` event is to provide insights into how frequently users experience issues.
It's not intended for logging or debugging.  

Fixes [#1778](https://github.com/WordPress/wordpress-playground/issues/1778)

## Testing Instructions (or ideally a Blueprint)

- The easiest way to see events is to add a `console.log` before this line https://github.com/WordPress/wordpress-playground/blob/acea593bc551aac14aebd982333be592ee4b298c/packages/playground/website/src/lib/tracking.ts#L26
- Open this link to [http://127.0.0.1:5400/website-server/?plugin=friends&theme=twentytwentythree](http://127.0.0.1:5400/website-server/?plugin=friends&theme=twentytwentythree)
- Confirm that a `installPlugin` and `installTheme` install event were sent with the slug and correct resource type
- Open this link to [install a plugin from a git resource.](http://127.0.0.1:5400/website-server/#{%22steps%22:%20[{%20%22step%22:%20%22installPlugin%22,%20%22pluginData%22:%20{%20%22resource%22:%20%22git:directory%22,%20%22url%22:%20%22https://github.com/WordPress/block-development-examples%22,%20%22ref%22:%20%22HEAD%22,%20%22path%22:%20%22plugins/data-basics-59c8f8%22%20},%20%22options%22:%20{%20%22activate%22:%20true%20}%20}]})
- Confirm that a `installPlugin` event was sent with the correct resource type
- Open this link [http://127.0.0.1:5400/website-server/?plugin=friefffnds&theme=astra](http://127.0.0.1:5400/website-server/?plugin=friefffnds&theme=astra)
- Confirm that at `bootSiteClient` error was sent
